### ISSUE TITLE
(Voicing) Lagging behaviour - what is it?

### ISSUE THREAD

**Issue by terracoda** [phetsims/john-travoltage#398]
It is hard to tell in interviews over zoom how much the lag is due to screen share and how much it has to do with sim.

In issue #397 , I think it might have to do with the actual design.

I opened this issue to note that I want to test performance once #397 is implemented.

**Comment by terracoda** [comment]
@jessegreenberg, could web speech just be a little laggy by nature?

**Comment by terracoda** [comment]
Actually, we can test performance once #397 and #399 are implemented.

**Comment by terracoda** [comment]
This issue does not have to be part of the milestone, but I don't know how to take it off the milestone.

**Comment by terracoda** [comment]
Using VoiceOver and no zoom, the sim doesn't seem too laggy to me.

**Comment by terracoda** [comment]
Hmm, @jessegreenberg, it might just be the focus events that a bit laggy. When I tab quickly through the sim (with a relatively fast speech rate), it is very easy to get 2 or 3 interactions behind.

**Comment by jessegreenberg** [comment]
This sounds familiar, and I think it will be extra laggy on Safari because of this workaround
```js
      // on safari, giving a bit of a delay to the speak request makes the `end`
      // SpeechSynthesisUtterance event come through much more consistently
      if ( platform.safari ) {
        stepTimer.setTimeout( () => {
          this.synth.speak( speechSynthUtterance );
        }, 500 );
      }
      else {
        this.synth.speak( speechSynthUtterance );
      }
```

So there will always be a 500 ms delay on that platform for every utterance. And that may interfere with the browsers ability to cancel speech as well. This is where Ill start investigating.

**Comment by terracoda** [comment]
Gotcha! Perhaps we no longer need the Safari delay since other improvements have been?
Keep me posted.

**Comment by terracoda** [comment]
I don't think the sim is lagging anymore. It is working great. Closing.
